packages/medusa/src/repositories/discount-condition.ts

```diff
@@ -108,39 +143,117 @@ export class DiscountConditionRepository extends Repository<DiscountCondition> {
   > {
     let toInsert: { condition_id: string; [x: string]: string }[] | [] = []
 
-    const { fromTable, resourceId } = this.getResourceIdentifiers(type)
+    const { conditionTable, joinTableForeignKey } =
+      this.getJoinTableResourceIdentifiers(type)
 
-    if (!fromTable || !resourceId) {
+    if (!conditionTable || !joinTableForeignKey) {
       return Promise.resolve([])
     }
 
     toInsert = resourceIds.map((pId) => ({
       condition_id: conditionId,
-      [resourceId]: pId,
+      [joinTableForeignKey]: pId,
     }))
 
     const insertResult = await this.createQueryBuilder()
       .insert()
       .orIgnore(true)
-      .into(fromTable)
+      .into(conditionTable)
       .values(toInsert)
       .execute()
 
     if (overrideExisting) {
       await this.createQueryBuilder()
         .delete()
-        .from(fromTable)
+        .from(conditionTable)
         .where({
           condition_id: conditionId,
-          [resourceId]: Not(In(resourceIds)),
+          [joinTableForeignKey]: Not(In(resourceIds)),
         })
         .execute()
     }
 
     return await this.manager
-      .createQueryBuilder(fromTable, "discon")
+      .createQueryBuilder(conditionTable, "discon")
       .select()
       .where(insertResult.identifiers)
       .getMany()
   }
+
+  async queryProductConditionTable({
+    type,
+    condId,
+    productId,
+  }): Promise<number> {
+    const { conditionTable, joinTable, joinTableForeignKey, resourceKey } =
+      this.getJoinTableResourceIdentifiers(type)
+
+    let productKey = "id"
+
+    if (type === DiscountConditionType.PRODUCT_TAGS) {
+      productKey = "product_id"
+    }
+
+    return await this.manager
+      .createQueryBuilder(conditionTable, "dc")
+      .innerJoin(
+        joinTable,
+        "resource",
+        `dc.${joinTableForeignKey} = resource.${resourceKey} and resource.${productKey} = :productId `,
+        {
+          productId,
+        }
+      )
+      .where(`dc.condition_id = :conditionId`, {
+        conditionId: condId,
+      })
+      .getCount()
```

Comment:
I've incorporated your suggested approach, and I really like it. One thing, that I'd like your take on is the naming of the different variables in use here. Do they make sense to you? @srindom 

cc @zakariaelas